docs/registration.md

```diff
@@ -383,21 +383,31 @@ All classes and functions documented here are directly importable from `textx` m
   parameters
 - `clear_language_registrations()` - deletes all languages registered
   programmatically. Note: languages registered through `setup.py` won't be removed
-- `metamodel_for_language(language_name)` - returns a meta-model for the given language name
+- `metamodel_for_language(language_name, **kwargs)` - returns a meta-model for
+  the given language name. `kwargs` are additional keyword arguments passed to
+  meta-model factory callable, similarly to `metamodel_from_str/file`.
 - `language_for_file(file_name_or_pattern)` - returns an instance of `LanguageDesc`
   for the given file name or file name pattern. Raises `TextXRegistrationError`
   if no language or multiple languages can parse the given file
 - `languages_for_file(file_name_or_pattern)` - returns a list of `LanguageDesc`
   for the given file name or file name pattern
-- `metamodel_for_file(file_name_or_pattern)` - returns a language meta-model for
-  a language that can parse the given file name or file name pattern. Raises
+- `metamodel_for_file(file_name_or_pattern, **kwargs)` - returns a language
+  meta-model for a language that can parse the given file name or file name
+  pattern. `kwargs` are additional keyword arguments passed to meta-model
+  factory callable, similarly to `metamodel_from_str/file`. Raises
   `TextXRegistrationError` if no language or multiple languages can parse the
   given file
 - `metamodels_for_file(file_name_or_pattern)` - returns a list of meta-models
   for languages that can parse the given file name or file pattern
 - `language` - a decorator used for [language registration](#registering-a-new-language)
 
 
+!!! warning
+
+    meta-model instances are cached so it is cheap to call
+    `metamodel_for_file/language` multiple times but `kwargs` used in the
+    first call only will be applied as the further calls return cached instance.
+
```

Comment:
Is it a good idea to *prevent using the cache* when passing some kwargs? 

This would allow to build the customized meta model once (with kwargs), the you can retrieve the registered language afterwards (w/o kwargs).

Reply:
Great idea. That would prevent buggy behavior if user would make a call with different kwargs while still retaining good performance in general case. I'll rework it to reinit meta-model whenever kwargs is not empty.

Reply:
Great!